core/src/main/java/org/eclipse/hono/config/quarkus/AuthenticatingClientOptions.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.config.quarkus;
+
+import java.util.Optional;
+
+import org.eclipse.hono.util.Constants;
+
+import io.smallrye.config.WithDefault;
+import io.smallrye.config.WithParentName;
+
+/**
+ * Options for configuring clients that need to authenticate to a remote server.
+ *
+ */
+public interface AuthenticatingClientOptions {
+
+    /**
+     * Gets the generic options.
+     *
+     * @return The options.
+     */
+    @WithParentName
+    GenericOptions genericOptions();
+
+    /**
+     * Gets the name or literal IP address of the host that the client is configured to connect to.
+     * <p>
+     * The default value of this property is <em>localhost</em>.
+     *
+     * @return The host name.
+     */
+    @WithDefault("localhost")
+    String host();
```

Comment:
@sophokles73 I found an issue here:
Setting the default to "localhost" will let `AuthenticatingClientConfigProperties#isHostConfigured()` always be `true`.

This means that if only Kafka is configured as messaging system, there will still be AMQP connection attempts.
I got corresponding connection attempt errors when deploying the Hono 1.10 SNAPSHOT components using the iot-packages hono helm chart with the 
`--set messagingNetworkTypes[0]=kafka --set kafkaMessagingClusterExample.enabled=true --set amqpMessagingNetworkExample.enabled=false`
params.

Reply:
I see. Do you want to fix it yourself or do you want me to do it? I am fine with both options ...

Reply:
Could you do it? I'm still working on the added metrics.

Reply:
Sure, will do ...

Reply:
I have pushed a8320ac1c59171984f1d612ff16eb319178ae24d which (hopefully) fixes the issue